src/Mod/PartDesign/PartDesignTests/TestTopologicalNamingProblem.py

```diff
@@ -0,0 +1,133 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# ***************************************************************************
+# *                                                                         *
+# *   Copyright (c) 2024 bgbsww@gmail.com                                   *
+# *                                                                         *
+# *   This file is part of FreeCAD.                                         *
+# *                                                                         *
+# *   FreeCAD is free software: you can redistribute it and/or modify it    *
+# *   under the terms of the GNU Lesser General Public License as           *
+# *   published by the Free Software Foundation, either version 2.1 of the  *
+# *   License, or (at your option) any later version.                       *
+# *                                                                         *
+# *   FreeCAD is distributed in the hope that it will be useful, but        *
+# *   WITHOUT ANY WARRANTY; without even the implied warranty of            *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU      *
+# *   Lesser General Public License for more details.                       *
+# *                                                                         *
+# *   You should have received a copy of the GNU Lesser General Public      *
+# *   License along with FreeCAD. If not, see                               *
+# *   <https://www.gnu.org/licenses/>.                                      *
+# *                                                                         *
+# ***************************************************************************
+import unittest
+
+
+import FreeCAD as App
+import Part
+import Sketcher
+import TestSketcherApp
+
+class TestTopologicalNamingProblem(unittest.TestCase):
+    def setUp(self):
+        self.Doc = App.newDocument("PartDesignTestTNP")
+
+    def testPadsOnBaseObject(self):
+        self.Body = self.Doc.addObject('PartDesign::Body','Body')
+        # Make first offset cube Pad
+        self.PadSketch = self.Doc.addObject('Sketcher::SketchObject', 'SketchPad')
+        self.Body.addObject(self.PadSketch)
+        TestSketcherApp.CreateRectangleSketch(self.PadSketch, (0, 1), (1, 1))
```

Comment:
The other 2 rectangles have the corner set at (0, 0).

Is it intentional to have this one's corner at (0, 1)?

Reply:
Good one.  That explains why I had to move the attachment point.  I'll adjust.